src/parser/expression.js

```diff
@@ -252,8 +252,16 @@ export default class ExpressionParser extends LValParser {
 
       if (update) {
         this.checkLVal(node.argument, undefined, undefined, "prefix operation");
-      } else if (this.state.strict && node.operator === "delete" && node.argument.type === "Identifier") {
-        this.raise(node.start, "Deleting local variable in strict mode");
+      } else if (this.state.strict && node.operator === "delete") {
+        const arg = node.argument;
+
+        if (arg.type === "Identifier") {
+          this.raise(node.start, "Deleting local variable in strict mode");
+        } else if (this.hasPlugin("classPrivateProperties")) {
+          if (arg.type === "PrivateName" || arg.property.type === "PrivateName") {
```

Comment:
Is the argument always guaranteed to be `Identifier`, `PrivateName` or `MemberExpression`?

If it happens to be neither `PrivateName` or `MemberExpression` in this branch it could throw `TypeError`.